### DISCUSSION TITLE
Getting X failed to open display, while testing GPU on the work-in-progress branch for the JetPack 6 Developer Preview

### DISCUSSION THREAD

**Discussion by PugalendhiMurugan** [OE4T/meta-tegra-community#112] [Q&A]
root@nvidia-jetson-agx-orin-devkit:/# /usr/bin/glmark2-es
-sh: /usr/bin/glmark2-es: No such file or directory
root@nvidia-jetson-agx-orin-devkit:/# /usr/bin/l4t-graphics-demos/x11/bubble
X failed to open display.
Usage: bubble [options]
  Frequency with which to automatically poke bubble:
    [-autopoke <frames>]
  Turn on framerate logging:
    [-fps]
  To run in Fixed-Frame Increment mode (Fixed increment of frames):
    [-ff]
  Delta used in Fixed-Frame Increment which defines amount of change in scene:
    [-delta <value>]
  Window system options (not all available on all platforms):
    [-windowsize] [<xres> <yres>]                  (window size)
    [-windowoffset <xpos> <ypos>]                  (window position)
                                   *Not supported on Wayland*

  Rendering options (availability varies by platform):
    [-buffering <n>]                               (n-buffered swaps)
                                                   (Max 4)
    [-msaa <samples>]                              (multi sampling)
    [-csaa <samples>]                              (coverage sampling)
    [-vcaa <samples>]                              (coverage sampling)
    [-useprogbin <boolean>]                        (program binary loading)
    [-smart <boolean>]                             (detect termination in
                                                   cross-partition stream)
    [-eglstreamsocket <socket name>]               (socket to get EGLStream from)
    [-fifo <length>]                               (0 min, 5 max)
    [-renderahead <length>]                        (-1 min, 10 max)
                                                   (Max number of in-flight GPU
                                                   frames in mailbox mode to
                                                   throttle mailbox mode.)
    [-latency <usec>]                              (0 min, 2147483647 max)
    [-timeout <usec>]                              (0 min, 2147483647 max)
    [-frames <#>]                                  (max numnber of frames to run)
    [-ip <IP address>]                             (IP address)
    [-port <int>]                                  (port number for multiple consumers)
    [-proctype <producer or consumer>]             (Producer or Consumer)
    [-surfaceid <n>]                               (0 min, 9999 max)
    [-sec <seconds>]                               (0 forever, 3153600 max)
    [-inactivity <secs>]                           (time to render on/off)

  Note:
    Use of parameters which modify the display configuration
    is primarily intended for use when running the application
    in a standalone fashion.  When there is a separate display
    server running, their use is discouraged, as they will blindly
    change the state without regard to any other applications
    sharing the display.
root@nvidia-jetson-agx-orin-devkit:/# /usr/bin/l4t-graphics-demos/x11/ctree
X failed to open display.
Usage: ctree [options]
  Put into demo mode:
    [-demo]
  Use low resolution textures:
    [-smalltex]
  Disable the menu interface:
    [-nomenu]
  Disable rendering of the sky:
    [-nosky]
  Turn on framerate logging:
    [-fps]
  Window system options (not all available on all platforms):
    [-windowsize] [<xres> <yres>]                  (window size)
    [-windowoffset <xpos> <ypos>]                  (window position)
                                   *Not supported on Wayland*

  Rendering options (availability varies by platform):
    [-buffering <n>]                               (n-buffered swaps)
                                                   (Max 4)
    [-msaa <samples>]                              (multi sampling)
    [-csaa <samples>]                              (coverage sampling)
    [-vcaa <samples>]                              (coverage sampling)
    [-useprogbin <boolean>]                        (program binary loading)
    [-smart <boolean>]                             (detect termination in
                                                   cross-partition stream)
    [-eglstreamsocket <socket name>]               (socket to get EGLStream from)
    [-fifo <length>]                               (0 min, 5 max)
    [-renderahead <length>]                        (-1 min, 10 max)
                                                   (Max number of in-flight GPU
                                                   frames in mailbox mode to
                                                   throttle mailbox mode.)
    [-latency <usec>]                              (0 min, 2147483647 max)
    [-timeout <usec>]                              (0 min, 2147483647 max)
    [-frames <#>]                                  (max numnber of frames to run)
    [-ip <IP address>]                             (IP address)
    [-port <int>]                                  (port number for multiple consumers)
    [-proctype <producer or consumer>]             (Producer or Consumer)
    [-surfaceid <n>]                               (0 min, 9999 max)
    [-sec <seconds>]                               (0 forever, 3153600 max)
    [-inactivity <secs>]                           (time to render on/off)

  Note:
    Use of parameters which modify the display configuration
    is primarily intended for use when running the application
    in a standalone fashion.  When there is a separate display
    server running, their use is discouraged, as they will blindly
    change the state without regard to any other applications
    sharing the display.
root@nvidia-jetson-agx-orin-devkit:/# /usr/bin/l4t-graphics-demos/x11/eglstreamcube
X failed to open display.
Usage: eglstreamcube [options] [command] [command options]
  Listen on <socket name>:
    [-socket <socket name>]
  Window system options (not all available on all platforms):
    [-windowsize] [<xres> <yres>]                  (window size)
    [-windowoffset <xpos> <ypos>]                  (window position)
                                   *Not supported on Wayland*

  Rendering options (availability varies by platform):
    [-buffering <n>]                               (n-buffered swaps)
                                                   (Max 4)
    [-msaa <samples>]                              (multi sampling)
    [-csaa <samples>]                              (coverage sampling)
    [-vcaa <samples>]                              (coverage sampling)
    [-useprogbin <boolean>]                        (program binary loading)
    [-smart <boolean>]                             (detect termination in
                                                   cross-partition stream)
    [-eglstreamsocket <socket name>]               (socket to get EGLStream from)
    [-fifo <length>]                               (0 min, 5 max)
    [-renderahead <length>]                        (-1 min, 10 max)
                                                   (Max number of in-flight GPU
                                                   frames in mailbox mode to
                                                   throttle mailbox mode.)
    [-latency <usec>]                              (0 min, 2147483647 max)
    [-timeout <usec>]                              (0 min, 2147483647 max)
    [-frames <#>]                                  (max numnber of frames to run)
    [-ip <IP address>]                             (IP address)
    [-port <int>]                                  (port number for multiple consumers)
    [-proctype <producer or consumer>]             (Producer or Consumer)
    [-surfaceid <n>]                               (0 min, 9999 max)
    [-sec <seconds>]                               (0 forever, 3153600 max)
    [-inactivity <secs>]                           (time to render on/off)

  Note:
    Use of parameters which modify the display configuration
    is primarily intended for use when running the application
    in a standalone fashion.  When there is a separate display
    server running, their use is discouraged, as they will blindly
    change the state without regard to any other applications
    sharing the display.
root@nvidia-jetson-agx-orin-devkit:/# /usr/bin/l4t-graphics-demos/x11/gears
X failed to open display.
Usage: gears [options] [runtime]
  (negative runtime means "forever")

  Clear color option:
    [-clearColor <r> <g> <b> <a>] (background color)

  Window system options (not all available on all platforms):
    [-windowsize] [<xres> <yres>]                  (window size)
    [-windowoffset <xpos> <ypos>]                  (window position)
                                   *Not supported on Wayland*

  Rendering options (availability varies by platform):
    [-buffering <n>]                               (n-buffered swaps)
                                                   (Max 4)
    [-msaa <samples>]                              (multi sampling)
    [-csaa <samples>]                              (coverage sampling)
    [-vcaa <samples>]                              (coverage sampling)
    [-useprogbin <boolean>]                        (program binary loading)
    [-smart <boolean>]                             (detect termination in
                                                   cross-partition stream)
    [-eglstreamsocket <socket name>]               (socket to get EGLStream from)
    [-fifo <length>]                               (0 min, 5 max)
    [-renderahead <length>]                        (-1 min, 10 max)
                                                   (Max number of in-flight GPU
                                                   frames in mailbox mode to
                                                   throttle mailbox mode.)
    [-latency <usec>]                              (0 min, 2147483647 max)
    [-timeout <usec>]                              (0 min, 2147483647 max)
    [-frames <#>]                                  (max numnber of frames to run)
    [-ip <IP address>]                             (IP address)
    [-port <int>]                                  (port number for multiple consumers)
    [-proctype <producer or consumer>]             (Producer or Consumer)
    [-surfaceid <n>]                               (0 min, 9999 max)
    [-sec <seconds>]                               (0 forever, 3153600 max)
    [-inactivity <secs>]                           (time to render on/off)

  Note:
    Use of parameters which modify the display configuration
    is primarily intended for use when running the application
    in a standalone fashion.  When there is a separate display
    server running, their use is discouraged, as they will blindly
    change the state without regard to any other applications
    sharing the display.
root@nvidia-jetson-agx-orin-devkit:/# /usr/bin/l4t-graphics-demos/x11/gearscube
X failed to open display.
Usage: gearscube [options]
    (negative runTime means "forever")
  Method to use for render-to-texture:
    [-method { fbo  pixmap  pbuffercopy }
  Texture size:
    [-texsize <size>]
  Window system options (not all available on all platforms):
    [-windowsize] [<xres> <yres>]                  (window size)
    [-windowoffset <xpos> <ypos>]                  (window position)
                                   *Not supported on Wayland*

  Rendering options (availability varies by platform):
    [-buffering <n>]                               (n-buffered swaps)
                                                   (Max 4)
    [-msaa <samples>]                              (multi sampling)
    [-csaa <samples>]                              (coverage sampling)
    [-vcaa <samples>]                              (coverage sampling)
    [-useprogbin <boolean>]                        (program binary loading)
    [-smart <boolean>]                             (detect termination in
                                                   cross-partition stream)
    [-eglstreamsocket <socket name>]               (socket to get EGLStream from)
    [-fifo <length>]                               (0 min, 5 max)
    [-renderahead <length>]                        (-1 min, 10 max)
                                                   (Max number of in-flight GPU
                                                   frames in mailbox mode to
                                                   throttle mailbox mode.)
    [-latency <usec>]                              (0 min, 2147483647 max)
    [-timeout <usec>]                              (0 min, 2147483647 max)
    [-frames <#>]                                  (max numnber of frames to run)
    [-ip <IP address>]                             (IP address)
    [-port <int>]                                  (port number for multiple consumers)
    [-proctype <producer or consumer>]             (Producer or Consumer)
    [-surfaceid <n>]                               (0 min, 9999 max)
    [-sec <seconds>]                               (0 forever, 3153600 max)
    [-inactivity <secs>]                           (time to render on/off)

  Note:
    Use of parameters which modify the display configuration
    is primarily intended for use when running the application
    in a standalone fashion.  When there is a separate display
    server running, their use is discouraged, as they will blindly
    change the state without regard to any other applications
    sharing the display.

**Comment by ichergui** [comment]
Hi @PugalendhiMurugan 

I tried all the `l4t-graphics-demos` `x11` sample applications. They work properly as expected.

Please avoid to  run those application on the serial UART or ssh session.

**Comment by ichergui** [comment]
BTW, could you please explain the origin of this application `glmark2-es` ? 
I couldn't find it in my build work directory.

**Reply by PugalendhiMurugan** [reply]
/usr/bin/glmark2-es

**Comment by PugalendhiMurugan** [comment]
root@nvidia-jetson-agx-orin-devkit:/# /usr/bin/glmark2-es2
Error: main: Could not initialize canvas
root@nvidia-jetson-agx-orin-devkit:/# /usr/bin/glmark2-es2-drm
Error: Failed to find a suitable connector
Error: main: Could not initialize canvas
root@nvidia-jetson-agx-orin-devkit:/# /usr/bin/glmark2-es2-wayland
Error: main: Could not initialize canvas
*********************
File names seems to different from my project directory....Tried executing them, failing to start due to above error.
